crates/eww/src/widgets/widget_definitions.rs

```diff
@@ -527,6 +527,38 @@ fn parse_icon_size(o: &str) -> Result<gtk::IconSize> {
     }
 }
 
+fn svg_to_pixbuf(
+    path: std::path::PathBuf,
+    image_width: i32,
+    image_height: i32,
+    fill: &str,
+) -> Result<Option<gtk::gdk_pixbuf::Pixbuf>> {
+    let svg_data: String = std::fs::read_to_string(path)?;
+
+    // The fastest way to add/change fill color
+    let svg_data = if svg_data.contains("fill=") {
+        let reg = regex::Regex::new(r#"fill="[^"]*""#)?;
+        reg.replace(&svg_data, &format!("fill=\"{}\"", fill))
+    } else {
+        let reg = regex::Regex::new(r"<svg")?;
+        reg.replace(&svg_data, &format!("<svg fill=\"{}\"", fill))
+    };
+
+    let pixbuf_svg = gtk::gdk_pixbuf::PixbufLoader::with_type("svg")?;
+
+    match (image_width, image_height) {
+        (w, h) if w > 0 || h > 0 => pixbuf_svg.set_size(w, h),
+        // Add default size to prevent widget overflowing, if image is too big
+        _ => pixbuf_svg.set_size(24, 24),
+    };
```

Comment:
```suggestion
    pixbuf_svg.set_size(image_width, image_height);
```
assuming defaults here might break configs relying on image sizes, i suggest removing this.

Reply:
The default values for image_width and image_height are -1, if left as default without adding the declarations, the image will cause the widget to grow and possibly fill the entire screen, resulting in the user no longer being able to manipulate the widget as SVGs tend to inherit size from the parent element if they do not have the width and height attributes.

Reply:
The default values ​​for image_width and image_height are -1, if left as default without adding the declarations, the image will cause the widget to grow and possibly fill the entire screen, resulting in the user no longer being able to manipulate the widget as SVGs tend to inherit or from viewBox sizes from the parent element if they do not have the width and height attributes.

Reply:
yes, but it'd still be a breaking change and why only implement this logic for svgs and not all images then?
also, why choose exactly 24*24 pixels?

Reply:
That's true, although currently the image-width and image-height attributes are not used for gif type images, which is wrong, but that would be in a separate PR, don't you think?

Reply:
That's true, although currently for gif type images the attributes image-width and image-height are not used, which is wrong, but that would be in a separate PR, don't you think?

And I used 24x24 to add a default size in case the attributes image-width/image-height are not defined, since otherwise some svg sizes like the github logo you attached are too big and the widget grows and overflows the screen

I currently added some logic to determine the size of the image in case any of the two attributes are defined, and thus display the image with its correct dimensions

Reply:
the attributes not being used with gifs is indeed a separate concern.
i'd rather add a notice to the docs to be careful about using large images instead of breaking the current behavior that people might rely on.
i don't think you understood my other point about the default dimensions
why 24*24 and not, for example, 300*300 as a default?
is there a reason for this?

Reply:
Yes, and it is to avoid the widget overflowing, the smaller the image dimensions the better, so we avoid future problems. Although you are right, maybe I will change it to 128x128 (which is the maximum standard recommended by most gnome icon packs, as far as I know), but if the image dimensions are less than 128x128 I will use the real dimensions, so there will be no problems.

I appreciate your patience

Reply:
> it might help to mark the attribute as optional
The image (regardless of the image type gif, webp, etc) is not displayed if I leave fill as optional, that's why I put an empty value as the default value. I'm even thinking that the best thing would be to create an exclusive widget for svg

Reply:
i managed to set it as optional without issues.
a separate might be an idea, but it should be for all kinds of vector graphics and i personally don't really see a reason for it

Reply:
How mark option as optional?

I did it this way but it doesn't work, i.e. the image without the fill attribute is not displayed.


```rs
prop(path: as_string, image_width: as_i32 = -1, image_height: as_i32 = -1, fill: as_string) {
```

Reply:
```rust
prop(path: as_string, image_width: as_i32 = -1, image_height: as_i32 = -1, fill: as_string?)
```

Reply:
Oh I understand now, thanks